src/targets/gpu/gemm_impl.cpp

```diff
@@ -129,7 +129,27 @@ auto rocblas_invoke(F f, Pack p, Ts... xs)
     });
 }
 
-static bool is_transposed(const shape& s) { return s.transposed() and s.strides().back() != 1; }
+static bool is_transposed(const shape& s)
+{
+    if(s.transposed())
+    {
+        return s.strides().back() != 1;
+    }
+    else if(s.strides() != s.as_standard().strides())
+    {
+        // If the strides are not standard, check if there's a permutation that produces those
+        // strides
+        auto lens_perm = s.lens();
+        std::sort(lens_perm.begin(), lens_perm.end());
```

Comment:
Why not call `find_permutation`? That should give the correct permutation without needing the loop.

Reply:
It seems like I misunderstood the implementation of `find_permutation` at first glance. I see now the purpose behind it. I changed the implementation accordingly.